src/headless/tasks/test/test_celery_task.py

```diff
@@ -393,3 +402,139 @@ def test_clean_metadata(self):
         self.assertTrue(isinstance(metadata['url'], basestring))
         self.assertTrue(
             isinstance(metadata['extra_keywords']['url'], basestring))
+
+    @unittest.skipUnless(
+        ast.literal_eval(os.environ.get('TASK_ALWAYS_EAGER', 'True')),
+        'This task only makes sense when it was tested on the same thread.'
+        'It will be skipped on async Celery test, because the validity check'
+        ' was on different QGIS Instance, thus will always be success '
+        '(and useless) on that setup.'
+    )
```

Comment:
Basically this test is not so useful on async mode, because it will always success (the caller and worker were on a different thread and different QGIS Instance)